### PR TITLE
Convert to typescript

### PR DESCRIPTION
**Pull request by rictic**

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+/**
+ * CleanKill hooks the interrupt handler, and provides callbacks for your code
+ * to cleanly shut down before the process exits.
+ */
+
+/**
+ * The type of a cleankill interrupt handler.
+ */
+export type Handler = (doneHandling: () => void) => void;
```
Comment (justinfagnani) [inline index.ts:22]: TODO: Convert to Promise?

File: index.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+export function onInterrupt(handler: Handler): void {
+  interruptHandlers.push(handler);
+}
+
+/**
+ * Call all interrupt handlers, and call the callback when they all complete.
+ *
+ * Removes the list of interrupt handlers.
+ */
+export function close(done: () => void): void {
```
Comment (justinfagnani) [inline index.ts:39]: TODO: Convert to Promise

File: index.ts (hunk 1 above)
Comment (rictic) [inline index.ts:22]: Done

File: index.ts (hunk 2 above)
Comment (nevir) [inline index.ts:39]: TODO: Convince every vendor to backport Promises

**Comment by justinfagnani** [comment]
LGTM

File: index.ts (hunk 2 above)
Comment (rictic) [inline index.ts:39]: We're moving to supporting only node ≥4 for our libraries. Converting to promises will be a major release though.

Comment (nevir) [inline index.ts:39]: 👍 just trolling :P

Comment (rictic) [inline index.ts:39]: Mind adding me as an owner in npm so I can push out a release?

Comment (notwaldorf) [inline index.ts:39]: Hi @nevir!!!!! @cdata says hi too 👋

Comment (justinfagnani) [inline index.ts:39]: TODO: Upgrade your node to something not ancient.

Comment (justinfagnani) [inline index.ts:39]: TODO: Get @nevir to upgrade his startup's legacy silo meta-framework mega-platform UI code to web components.

Comment (nevir) [inline index.ts:39]: @notwaldorf @cdata Heya! :D

> TODO: Get @nevir to upgrade his startup's legacy silo meta-framework mega-platform UI code to web components.

TODO: progressive-webify legacy meta-framework mega-platform

Comment (nevir) [inline index.ts:39]: (on the flip side, we are now well on our path to being a typescript-only shop - it feels gooood)

**Pull request merged**